packages/caravan-wallets/src/index.ts

```diff
@@ -335,6 +359,12 @@ export function SignMultisigTransaction({
   progressCallback,
 }: SignMultisigTransactionArgs) {
   switch (keystore) {
+    case BITBOX:
+      return new BitBoxSignMultisigTransaction({
+        walletConfig,
+        psbt,
```

Comment:
It looks like the `psbt` passed here is actually undefined. Short of us applying a proper fix, a quick hack is to do what's done below with ledger
```js
@@ -360,9 +360,18 @@ export function SignMultisigTransaction({
 }: SignMultisigTransactionArgs) {
   switch (keystore) {
     case BITBOX:
+      let _psbt = psbt;
+      if (!_psbt)
+        _psbt = getUnsignedMultisigPsbtV0({
+          network,
+          inputs: inputs ? inputs.map(convertLegacyInput) : [],
+          outputs: outputs ? outputs.map(convertLegacyOutput) : [],
+        }).toBase64();
       return new BitBoxSignMultisigTransaction({
         walletConfig,
-        psbt,
+        psbt: _psbt,
         returnSignatureArray,
       });
```

Tried this on my machine with a P2WSH PSBT and and got stuck at [Loading transaction...](https://github.com/BitBoxSwiss/bitbox02-firmware/blob/2c737961e830ba2dd1b2288c7b8211ef71ae4092/src/rust/bitbox02-rust/src/hww/api/bitcoin/signtx.rs#L591) when running [`pairedBitBox.btcSignPSBT`](https://github.com/BitBoxSwiss/bitbox-api-rs/blob/7abaa32aafbea466f4330a8859b318ade4f536c1/src/wasm/mod.rs#L342)

<img width="1256" alt="image" src="https://github.com/user-attachments/assets/096ddf90-81a8-47a0-8aa8-ddd922c29495">
[samplepsbt.txt](https://github.com/user-attachments/files/16829714/samplepsbt.txt)

Reply:
Note this repo needs a fix for segwit PSBTs and this is my current hack
[cc_p2wsh_fix.patch](https://github.com/user-attachments/files/16829754/cc_p2wsh_fix.patch)

Reply:
@chadchapnick thanks for the review and for testing.

As for the issue you saw here, this is because the PSBT is missing the non_witness_utxo field in the inputs. See this comment from the commit message:

> - The BitBox02 requires the previous transaction of each input to be
present in the PSBT (`PSBT_IN_NON_WITNESS_UTXO`), so it can verify the
input amount and avoid fee attacks. The signing test fixtures are
missing these, so they fail.

@bucko13 mentioned to me that this is only a problem with the test fixtures, but that you do supply this field in production.

Reply:
@benma thank you for the clarification! Can confirm signing on device works very smoothly now for both P2WSH and P2SH-P2WSH. Fascinating to peer into the stack you folks have architected.

Reply:
I applied the workaround for the undefined psbt like Ledger as you suggested

Reply:
We're going to fix this upstream so this hack can actually be dropped, and we can use what you had before—thank you for entertaining it 
```js
BitBoxSignMultisigTransaction({
         walletConfig,
         psbt,
         returnSignatureArray,
       });
```